package.json

```diff
@@ -16,20 +16,19 @@
   },
   "dependencies": {
     "addressparser": "^0.3.2",
-    "emailjs-mime-codec": "^2.0.5",
-    "moment": "2.20.1",
-    "starttls": "1.0.1"
+    "emailjs-mime-codec": "^2.0.7",
```

Comment:
before there was a yarn/package.json lock file, i had some experience with even minor bumps introducing bugs that impacted deployments that did not intend to have anything upgraded.

now with locks, i don't see that happen as much. what are you thoughts on it? for dev dependencies, i don't think it matters as much, but for direct dependencies, it's worth thinking about.

since you removed moment and changed startls to be 'compatible' they are now all consistent. so that is good.

let me know what you think and we can run with this or to be safe we can change them all to be locked to one version.

Reply:
personally my stance is, now that lockfiles are normalized, an end user can and should be able to ensure their builds are reproducible without libraries needing to pin themselves -- in fact, libraries should avail themselves of all the latest versions of libraries they can take without breaking tests. that's the best way (in my mind) to ensure the most secure and bug-free code is shipped.

that being said, latest version of node 6 ships with npm 3.10.3, and package-lock wasn't introduced until npm 5 -- i wouldn't want to dismiss people in that situation out-of-hand, knowing that e.g. [joyent will still be on 0.10 for some time](https://github.com/joyent/node-asn1/pull/30#discussion_r193832592).

i wonder, though, if people in that situation would even bother touching their dependencies, especially if the next release is cut as semver major. it might be a moot point altogether, in which case i fall back to my original argument.